### PR TITLE
House: Chaining the models City Country

### PR DESCRIPTION
**Pull request by shlior7**
Relevant Issues: #114
I removed the countries selection logic from the HTML file, and replace it by the lib "[django-smart-select](https://django-smart-selects.readthedocs.io/en/latest/)" 

this lib chains models with jquery fetching automatically in the admin and in any form as seen in the docs.

the latest version of this lib[(1.6.0)](https://github.com/jazzband/django-smart-selects/blob/master/CHANGES.rst) is the first supporting python 3.9 which we are using but this version isn't released yet. so the solution i saw people use is to explicitly state that we are using the lib from the version of the repo in github see Pipfile.

and thus no more worrying about logic in the HTML.

Because all this magic is happening from the 3rd party library there is no real code to test that isn't tested thoroughly in the library itself, so I don't think any new tests need to be created If there is I would like to hear the idea of the test because the output of this PR is still "frontendy" meaning changes are happening to the HTML element "select" and its options so I cant think of a test that tests the code behind that. But I checked this on the House filter form and in the admin and this works perfectly 

The tests changed only because we're creating the initial countries and cities data using the django command loaddata, so the tests need to run that first as a fixture. and that created some problems with the test classes so I moved the problematic ones to be functional and that fixed the problem.

https://user-images.githubusercontent.com/34008861/168373694-0c1ed856-e53a-4873-9f27-9e774aed67c8.mov

https://user-images.githubusercontent.com/34008861/168373920-ef21610d-0757-494f-ad58-f90a673014f5.mov

### CODE REVIEW

**Comment by shlior7** [comment]
Had a problem with the last #115, made a copy of this and removed some "formatting" changes and fixed the empty line in EOF
@adiabramovitch

**Comment by adiabramovitch** [comment]
Please add information in order to have smart-select in a local env.

**Comment by shlior7** [comment]
> Please add information in order to have smart-select in a local env.

Im not sure I understand, there are no steps to have the library in a local env, pipenv install should do it no?

**Comment by adiabramovitch** [comment]
For some reason it does not work locally, it failed so thought you did something specific. 
Can someone else check that it does not break the local env end and that there is no need for further configuration?
@ronyyosef @LiorKGOW @ShellyGolden @Yoavsc129 @orIsrael

File: templates/global/filter_form.html [hunk 1]
```diff
@@ -4,10 +4,10 @@ <h3 class="text-center text-black pt-1">House Filter</h3>
     <form
             method="post"
             id="HouseFilter"
-            data-cities-url="{% url 'house:ajax_load_cities' %}"
             novalidate
     >
         <table>
+            {{ form.media.js }}
```
Comment (adiabramovitch) [inline templates/global/filter_form.html:10]: What is the media.js file?

Comment (shlior7) [reply templates/global/filter_form.html:10]: these are external scripts that the library added on the form so it would be reactive to the changes of the select elements
it is documented here :https://django-smart-selects.readthedocs.io/en/latest/usage.html#usage-in-templates

File: house/views.py [hunk 2]
```diff
@@ -46,29 +46,24 @@ def house_view(request):
...
+                user=request.user,
+                name=cleaned_data['name'],
+                public=cleaned_data['public'],
+                parent_profession_1=cleaned_data['parent_profession_1'],
+                parent_profession_2=cleaned_data['parent_profession_2'],
+                country=cleaned_data['country'],
+                city=cleaned_data['city'],
+                children=cleaned_data['children'],
+                income=cleaned_data['income'],
+            )
```
Comment (adiabramovitch) [inline house/views.py:66]: Is that relevant? isn't it just formatting
If so, all related formatting changes should be in a different PR.

File: house/models.py [hunk 3]
```diff
@@ -49,17 +50,25 @@ class House(models.Model):
...
     income = models.IntegerField(null=True, blank=True, validators=[MinValueValidator(0)])
     children = models.IntegerField(null=True, blank=True, validators=[MinValueValidator(0)])
     description = models.TextField(max_length=250, default='')
 
     @staticmethod
-    def create_house(user, name, public, country, city, parent_profession_1, parent_profession_2, income, children,
-                     description=''):
+    def create_house(
+        user, name, public, country, city, parent_profession_1, parent_profession_2, income, children, description=''
+    ):
```
Comment (adiabramovitch) [inline house/models.py:71]: ditto

File: house/models.py [hunk 4]
```diff
@@ -70,7 +79,7 @@ def create_house(user, name, public, country, city, parent_profession_1, parent_
             parent_profession_2=parent_profession_2,
             income=income,
             children=children,
-            description=description
+            description=description,
```
Comment (adiabramovitch) [inline house/models.py:82]: ditto

File: expenses/models.py [hunk 5]
```diff
@@ -37,7 +37,10 @@ def create_expense(house_name, amount, date, category):
...
     def average_expenses_of_houses_by_categories(houses) -> QuerySet:
-        filtered_expenses = Expenses.objects.filter(
-            house_name__user__in=houses.values_list('user')).order_by().values(
-            'category').annotate(average=Avg("amount", output_field=IntegerField()))
+        filtered_expenses = (
+            Expenses.objects.filter(house_name__user__in=houses.values_list('user'))
+            .order_by()
+            .values('category')
+            .annotate(average=Avg("amount", output_field=IntegerField()))
+        )
```
Comment (adiabramovitch) [inline expenses/models.py:45]: ditto

File: tests/conftest.py [hunk 6]
```diff
@@ -1,13 +1,27 @@
...
+    HOUSE_DESCRIPTION,
+    EXPENSE_AMOUNT,
+    EXPENSE_DATE,
+    EXPENSE_CATEGORY,
+    USERNAME,
+    PASSWORD,
+    FIRSTNAME,
+    LASTNAME,
+    EMAIL,
+)
```
Comment (adiabramovitch) [inline tests/conftest.py:24]: ditto

**Comment by adiabramovitch** [comment]
Please move all the formatting changes into  a different PR and we will merge both of them now

**Comment by shlior7** [comment]
@adiabramovitch 
> Please move all the formatting changes into a different PR and we will merge both of them now 

created a new PR #134 both ready to merge whenever you're ready
